__react/template/src/index.ts.js

```diff
@@ -0,0 +1,143 @@
+          subscribeMessage,
+          publishMessage,
+          channel.description(),
+          channel.parameters(),
+          params
+        );
+      }
+    }
+
+    if (isPubsub(channel)) {
+      if (channel.hasSubscribe()) {
+        return Publish(
+          asyncapi.defaultContentType(), 
+          channelName, 
+          subscribeMessage, 
+          channel.description(), 
+          channel.parameters());
+      }
+      if (channel.hasPublish()) {
+        return Subscribe(
+          asyncapi.defaultContentType(), 
+          channelName, 
+          publishMessage, 
+          channel.description(), 
+          channel.parameters());
+      }
+    }
+  }) : '';
+  return channelWrappers;
+}
+
+
+export default function index({ asyncapi, params }) {
+  //Import the channel and messages and re-export them
+  const importList = [];
+  const exportList = [];
+  for (const [channelName] of Object.entries(asyncapi.channels())){
+    const camelCaseChannelName = camelCase(channelName);
+    importList.push(`import * as ${camelCaseChannelName}Channel from "./channels/${pascalCase(channelName)}";`);
+    exportList.push(`export {${camelCaseChannelName}Channel};`);
+  }
+
+  //Import the messages and re-export them
+  for (const [messageName] of asyncapi.allMessages()) {
+    const pascalMessageName = pascalCase(messageName);
+    importList.push(`import * as ${pascalMessageName}Message from "./messages/${pascalMessageName}";`);
+    exportList.push(`export {${pascalMessageName}Message};`);
+  }
+
+  return (
+    <File name="index.ts">
+{`
+import {fromSeed} from 'ts-nkeys';
+import {AvailableHooks, receivedDataHook, BeforeSendingDataHook, Hooks} from './hooks';
+export {AvailableHooks, receivedDataHook, BeforeSendingDataHook, Hooks}
+import * as TestClient from './tests/testclient/';
+export {TestClient};
+import {ErrorCode, NatsTypescriptTemplateError} from './NatsTypescriptTemplateError';
+export {ErrorCode, NatsTypescriptTemplateError}
+import { 
+  Client, 
+  NatsConnectionOptions, 
+  connect,
+  Payload, 
+  NatsError, 
+  Subscription, 
+  ServersChangedEvent, 
+  SubEvent, 
+  ServerInfo,
+  SubscriptionOptions
+  } from 'ts-nats';
+  
+export {Client, ServerInfo, ServersChangedEvent, SubEvent}
+
+${importList.join('')}
+${exportList.join('')}
+
+import * as events from 'events';
+export enum AvailableEvents {
+  permissionError = 'permissionError',
+  close = 'close',
+  connect = 'connect',
+  connecting = 'connecting',
+  disconnect = 'disconnect',
+  error = 'error',
+  pingcount = 'pingcount',
+  pingtimer = 'pingtimer',
+  reconnect = 'reconnect',
+  reconnecting = 'reconnecting',
+  serversChanged = 'serversChanged',
+  subscribe = 'subscribe',
+  unsubscribe = 'unsubscribe',
+  yield = 'yield'
+}
+
+export declare interface NatsAsyncApiClient {
+  ${Events()}
+}
+
+export class NatsAsyncApiClient extends events.EventEmitter{
+  ${Standard(asyncapi)}
+  ${getChannelWrappers(asyncapi, params).join('')}
+}
```

Comment:
I meaned imports and exports groups in this whole block :)

Reply:
ahh gotcha, changing.